modules/luci-mod-network/htdocs/luci-static/resources/view/network/interfaces.js

```diff
@@ -765,19 +765,32 @@ return L.view.extend({
 								if (nameval == null || protoval == null || nameval == '' || protoval == '')
 									return;
 
-								return m.save(function() {
-									var section_id = uci.add('network', 'interface', nameval);
+								return network.getProtocol(protoval).isCreateable(nameval).then(L.bind(function(checkval) {
```

Comment:
- Please cache the result of `network.getProtocol(protoval)` in a local variable (e.g. `protoclass`)
- No need to `bind()` the `then()` function since we do not use `this` anywhere